docs/high-availability.md

```diff
@@ -0,0 +1,196 @@
+done.
+
+Clustered etcd already replicates your storage to all master instances in your cluster.  This means that to lose data, all three nodes would need
+to have their physical (or virtual) disks fail at the same time.  The probability that this occurs is relatively low, so for many people
+running a replicated etcd cluster is likely reliable enough.  You can add additional reliability by increasing the
+size of the cluster from three to five nodes.
+
+### Even more reliable storage
+Of course, if you are interested in increased data reliability, there are further options (belts *and* suspenders, ftw!).
+
+If you use a cloud provider, then they usually provide this
+for you, for example [Persistent Disk](https://cloud.google.com/compute/docs/disks/persistent-disks) on the Google Cloud Platform.  These
+are block-device persistent storage that can be mounted onto your virtual machine. Other cloud providers provide similar solutions.
+
+If you are running on physical machines, you can also use network attached redundant storage using an iSCSI or NFS interface.
+Alternatively, you can run a clustered file system like Gluster or Ceph.  Finally, you can also run a RAID array on each physical machine.
+
+Regardless of how you choose to implement it, if you chose to use one of these options, you should make sure that your storage is mounted
+to each machine.  If your storage is shared between the three masters in your cluster, you should create a different directory on the storage
+for each node.  Throughout these instructions, we assume that this storage is mounted to your machine in ```/var/etcd/data```
+
+## Reliable nodes
+On each master node, we are going to run a number of processes that implement the Kubernetes API.  The first step in making these reliable is
+to make sure that each automatically restarts when it fails.  To achieve this, we need to install a process watcher.  We choose to use
+the ```kubelet``` that we use on each of the nodes.  This is convenient, since we can use containers to distribute our binaries, we can
+establish resource limits, and introspect the resource usage of each daemon.  Of course, we also need something to monitor the kubelet
+itself (insert who watches the watcher jokes here).  To do this, we'll use the ```monit``` tool, but there are a number of alternate
+choices.  We also need to health check and restart the docker daemon, we'll use ```monit``` for this as well.
+
+You should install the [kubelet binary](https://storage.googleapis.com/kubernetes-release/release/v0.18.0/bin/linux/amd64/kubelet), the
+[/etc/init.d/kubelet](high-availability/init-kubelet) and [/etc/default/kubelet](high-availability/default-kubelet)
+scripts.  You should also install the monit daemon (```apt-get install monit```) and the [/etc/monit/conf.d/kubelet](high-availability/monit-kubelet) and
+[/etc/monit/conf.d/docker](high-availability/monit-docker) configs.
```

Comment:
suggestion -
"
...first step make these reliable...need a watcher process.

for debian, we choose monit....

for systemd-based systems, you can run 'systemctl enable kubelet'. you should also have run 'systemctl enable docker'
"